pages/works/[id].tsx

```diff
@@ -0,0 +1,110 @@
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import styles from "@/styles/WorkSinglePage.module.css";
+import { fontSyne, fontSatoshi } from "../_app";
+import works from "@/constants/works.json";
+import Divider from "@/components/base/Divider";
+import Contact from "@/components/general/Contact";
+import InfoItem from "@/components/general/InfoItem";
+
+const WorkSinglePage = () => {
+  const [data, setData] = useState({
+    id: "",
+    title: "",
+    intro: "",
+    info: {
+      client: "",
+      timeline: "",
+      service: "",
+      website: "",
+    },
+    report: {
+      challenge: "",
+      solution: "",
+      result: "",
+    },
+    img: [
+      {
+        src: "",
+        alt: "",
+      },
+      {
+        src: "",
+        alt: "",
+      },
+      {
+        src: "",
+        alt: "",
+      },
+    ],
+  });
+
+  const router = useRouter();
+  const id = router.query.id;
+
+  //which is better? load data here or load it when initiate state
```

Comment:
load data inside of useEffect is better because
1. the page can render something to show to the users
2. after init render, load data and reset state, re-render page based on the state changes.
3. if need a backend or API to fetch the data, may not need to change too much.

Reply:
Got it, thanks